examples/classification/plot_lda_qda.py

```diff
@@ -144,54 +103,122 @@ def plot_ellipse(splot, mean, cov, color):
+import matplotlib.pyplot as plt
 
 from sklearn.discriminant_analysis import (
     LinearDiscriminantAnalysis,
     QuadraticDiscriminantAnalysis,
 )
 
-for i, (X, y) in enumerate([dataset_fixed_cov(), dataset_cov()]):
-    # Linear Discriminant Analysis
-    lda = LinearDiscriminantAnalysis(solver="svd", store_covariance=True)
-    y_pred = lda.fit(X, y).predict(X)
-    splot = plot_data(lda, X, y, y_pred, fig_index=2 * i + 1)
-    plot_lda_cov(lda, splot)
-    plt.axis("tight")
-
-    # Quadratic Discriminant Analysis
-    qda = QuadraticDiscriminantAnalysis(store_covariance=True)
-    y_pred = qda.fit(X, y).predict(X)
-    splot = plot_data(qda, X, y, y_pred, fig_index=2 * i + 2)
-    plot_qda_cov(qda, splot)
-    plt.axis("tight")
-
-plt.tight_layout()
-plt.subplots_adjust(top=0.92)
+fig, axs = plt.subplots(nrows=3, ncols=2, sharex="row", sharey="row", figsize=(8, 12))
+
+lda = LinearDiscriminantAnalysis(solver="svd", store_covariance=True)
+qda = QuadraticDiscriminantAnalysis(store_covariance=True)
+
+for ax_row, X, y in zip(
+    axs,
+    (X_isotropic_covariance, X_shared_covariance, X_different_covariance),
+    (y_isotropic_covariance, y_shared_covariance, y_different_covariance),
+):
+    lda.fit(X, y)
+    plot_result(lda, X, y, ax_row[0])
+    qda.fit(X, y)
+    plot_result(qda, X, y, ax_row[1])
+
+axs[0, 0].set_title("Linear Discriminant Analysis")
+axs[0, 0].set_ylabel("Data with fixed and spherical covariance")
+axs[1, 0].set_ylabel("Data with fixed covariance")
+axs[0, 1].set_title("Quadratic Discriminant Analysis")
+axs[2, 0].set_ylabel("Data with varying covariances")
+fig.suptitle(
+    "Linear Discriminant Analysis vs Quadratic Discriminant Analysis",
+    y=0.94,
+    fontsize=15,
+)
 plt.show()
+
+# %%
+# The first important thing to notice is that both LDA and QDA are equivalent for the
+# first and second datasets. Indeed, the major difference between LDA and QDA is that
+# LDA assumes that the covariance matrix of each class is equal, while QDA will estimate
+# a covariance matrix for each class. Since the data generative process used the same
+# covariance matrix for both classes, QDA estimated two covariance matrices that are
+# equal and therefore equivalent to the covariance matrix estimated by LDA.
+#
+# In the first dataset, we see that the discriminant boundary corresponds to the
+# perpendicular bisector between the two means. This is the case because the covariance
+# matrix used to generate the dataset is spherical. Therefore, we observe that the
+# discriminant boundary in the second dataset is no longer a perpendicular bisector.
+# It only passes through the middle of the two means.
+#
+# Finally, in the third dataset, we observe the real difference between the modeling of
+# LDA and QDA. QDA is fitting two covariance matrices and provides a non-linear
+# discriminant boundary, while LDA is underfitting since it assumes that a single
+# covariance matrix is shared by both classes.
```

Comment:
```suggestion
# The first important thing to notice is that LDA and QDA are equivalent for the
# first and second datasets. Indeed, the major difference is that LDA assumes
# that the covariance matrix of each class is equal, while QDA estimates a
# covariance matrix per class. Since in these cases the data generative process
# has the same covariance matrix for both classes, QDA estimates two covariance
# matrices that are (almost) equal and therefore equivalent to the covariance
# matrix estimated by LDA.
#
# In the first dataset the covariance matrix used to generate the dataset is
# spherical, which results in a discriminant boundary that aligns with the
# perpendicular bisector between the two means. This is no longer the case for
# the second dataset. The discriminant boundary only passes through the middle
# of the two means.
#
# Finally, in the third dataset, we observe the real difference between LDA and
# QDA. QDA fits two covariance matrices and provides a non-linear discriminant
# boundary, whereas LDA underfits since it assumes that both classes share a
# single covariance matrix.
```